### DISCUSSION TITLE
How to build framework for IOS

### DISCUSSION THREAD

**Discussion by AnitaGVMTechnologies** [v2fly/v2ray-core#2882] [General]
Hello there
Is anybody has idea , how we can create framework for ios using v2ray 

I am working on IOS app for learning purpose and i need v2Ray support in my IOS app
Can anybody help me here

**Comment by alexjane19** [comment]
you should use gomobile to build v2ray source code from this repository for ios. You can find some examples on README from this repo: https://github.com/XTLS/Xray-core

**Comment by AnitaGVMTechnologies** [comment]
could you please send me specifi link , there are so many

On Fri, Feb 9, 2024 at 2:00 PM Alex Jane ***@***.***> wrote:

> you should use gomobile to build v2ray source code from this repository
> for ios. You can find some examples on README from this repo:
> https://github.com/XTLS/Xray-core
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/v2fly/v2ray-core/discussions/2882#discussioncomment-8416186>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7ZM6FRPMGP47CFE2LPUXD3YSXNB7AVCNFSM6AAAAABDBBAMESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMJWGE4DM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards,
Anita Nagori
Project Manager |  GVM Technologies LLP
Skype – anita4gvm
***@***.***
http://www.gvmtechnologies.com


The content of this email is confidential and intended for the recipient
specified in the message only. It is strictly forbidden to share any part
of this message with any third party, without the written consent of the
sender. If you received this message by mistake, please reply to this
message and follow with its deletion, so that we can ensure such a mistake
does not occur in the future.

**Reply by alexjane19** [reply]
under GUI-clients section, you can find iOS implementations, 
https://github.com/XTLS/Xray-core#gui-clients

**Comment by AnitaGVMTechnologies** [comment]
one of them are not found and other 2 are appstore links , here i am
searching for building framework for v2ray using v2core plugin , i need
steps which help me to build framework which i can place in my ios project
and work with v2ray

On Fri, Feb 9, 2024 at 3:10 PM Alex Jane ***@***.***> wrote:

> under GUI-clients section, you can find iOS implementations,
> https://github.com/XTLS/Xray-core#gui-clients
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/v2fly/v2ray-core/discussions/2882#discussioncomment-8416847>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7ZM6FWUEC7YWKOGKOZCZWTYSXVKVAVCNFSM6AAAAABDBBAMESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMJWHA2DO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards,
Anita Nagori
Project Manager |  GVM Technologies LLP
Skype – anita4gvm
***@***.***
http://www.gvmtechnologies.com


The content of this email is confidential and intended for the recipient
specified in the message only. It is strictly forbidden to share any part
of this message with any third party, without the written consent of the
sender. If you received this message by mistake, please reply to this
message and follow with its deletion, so that we can ensure such a mistake
does not occur in the future.

**Reply by alexjane19** [reply]
https://github.com/dxdx123/Clash-for-iOS
https://github.com/amir-devman/Mango?tab=readme-ov-file

you should read one of this repo to know how you can achieve in your code. That's what you need

**Reply by laichunhui** [reply]
XrayKit is miss, Do you still have the source code locally?

**Comment by AnitaGVMTechnologies** [comment]
This is asking for github token and id , i have given but it keep failed it
even though it shown connected

On Fri, Feb 9, 2024 at 3:32 PM Alex Jane ***@***.***> wrote:

> https://github.com/dxdx123/Clash-for-iOS
> https://github.com/amir-devman/Mango?tab=readme-ov-file
>
> you should read one of this repo to know how you can achieve in your code.
> That's what you need
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/v2fly/v2ray-core/discussions/2882#discussioncomment-8417074>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7ZM6FUPARLJ55FXONWQQD3YSXX2NAVCNFSM6AAAAABDBBAMESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMJXGA3TI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards,
Anita Nagori
Project Manager |  GVM Technologies LLP
Skype – anita4gvm
***@***.***
http://www.gvmtechnologies.com


The content of this email is confidential and intended for the recipient
specified in the message only. It is strictly forbidden to share any part
of this message with any third party, without the written consent of the
sender. If you received this message by mistake, please reply to this
message and follow with its deletion, so that we can ensure such a mistake
does not occur in the future.

**Comment by AnitaGVMTechnologies** [comment]
is Tun2SocksKit and XrayKit support v2Ray, vless , vmess , trojan etc ?

On Fri, Feb 9, 2024 at 3:51 PM Anita Nagori ***@***.***>
wrote:

> This is asking for github token and id , i have given but it keep failed
> it even though it shown connected
>
> On Fri, Feb 9, 2024 at 3:32 PM Alex Jane ***@***.***> wrote:
>
>> https://github.com/dxdx123/Clash-for-iOS
>> https://github.com/amir-devman/Mango?tab=readme-ov-file
>>
>> you should read one of this repo to know how you can achieve in your
>> code. That's what you need
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/v2fly/v2ray-core/discussions/2882#discussioncomment-8417074>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A7ZM6FUPARLJ55FXONWQQD3YSXX2NAVCNFSM6AAAAABDBBAMESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMJXGA3TI>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Thanks & Regards,
> Anita Nagori
> Project Manager |  GVM Technologies LLP
> Skype – anita4gvm
> ***@***.***
> http://www.gvmtechnologies.com
>
>
> The content of this email is confidential and intended for the recipient
> specified in the message only. It is strictly forbidden to share any part
> of this message with any third party, without the written consent of the
> sender. If you received this message by mistake, please reply to this
> message and follow with its deletion, so that we can ensure such a mistake
> does not occur in the future.
>


-- 
Thanks & Regards,
Anita Nagori
Project Manager |  GVM Technologies LLP
Skype – anita4gvm
***@***.***
http://www.gvmtechnologies.com


The content of this email is confidential and intended for the recipient
specified in the message only. It is strictly forbidden to share any part
of this message with any third party, without the written consent of the
sender. If you received this message by mistake, please reply to this
message and follow with its deletion, so that we can ensure such a mistake
does not occur in the future.

**Comment by ghost** [comment]
tell me your email

**Reply by hassanAzhar143** [reply]
hassan@mobizion.com

**Reply by hassanAzhar143** [reply]
I need the solution for this

**Reply by laichunhui** [reply]
Hello, have you achieved your goal? Can you share it?

**Reply by hassanAzhar143** [reply]
<img width="847" alt="Screenshot 2024-05-17 at 3 37 05 PM" src="https://github.com/v2fly/v2ray-core/assets/150673894/000da491-bd5d-45df-8d6e-f7b8890e6214">
These are the required libraries,
but unfortunatly i could not figure it out until now,my DNS is making truble

**Reply by hassanAzhar143** [reply]
2024/05/17 11:05:09 [Warning] [1294391031] app/proxyman/outbound: failed to process outbound traffic > proxy/vmess/outbound: failed to find an available destination > common/retry: [dial tcp 11.11.11.5:8080: i/o timeout dial tcp: lookup Domain.com: operation was canceled] > common/retry: all retry attempts failed

**Reply by hassanAzhar143** [reply]
@laichunhui
 , do you find any resource?

**Reply by laichunhui** [reply]
not have, many open source libraries are missing key dependencies.

**Reply by ghost** [reply]
我有，不过收费，我自己用rust写的

**Reply by ghost** [reply]
如果你不用vmess可以用leaf

**Reply by hassanAzhar143** [reply]
I need ios solution,for VMESS tcp tls.

**Reply by themoein100** [reply]
> 如果你不用vmess可以用leaf

你好，我对你用Rust写的VMess+TLS核心很感兴趣。请问可以提供一个联系方式吗？我想了解购买的方式和价格。谢谢！
